data/src/main/java/com/example/data/database/MovieRepositoryImpl.kt

```diff
@@ -23,4 +24,19 @@ class MovieRepositoryImpl(private val movieDao: MovieDao) : MovieRepository {
             movieDao.insertNowPlayingMovie(it.transformToMovieEntity())
         }
     }
+
+    override fun getTopRatedMovies(): Result<List<Movie>> =
```

Comment:
these 2 methods are the same that the methods above. You can abstract the functionality by passing the movie type by param and reusing the same methods